docs/source/release_notes.rst

```diff
@@ -10,12 +10,13 @@ Future Release
     * Changes
         * Remove lines in ``LogicalType.transform`` that raise error if dtype conflicts (:pr:`1012`)
         * Add ``infer_datetime_format`` param to speed up ``to_datetime`` calls (:pr:`1016`)
+        * Adds datetime format inference in transform if a format is not set (:pr:`1025`)
```

Comment:
I think I'd call this an enhancement instead of a change and say something like `Store inferred datetime format on Datetime logical type instance`